cachi2/core/package_managers/pip.py

```diff
@@ -1265,7 +1264,7 @@ def copy(
         requirement.raw_package = self.raw_package
         # Extras are incorrectly treated as part of the URL itself. If we're setting
         # the URL, clear them.
-        requirement.extras = [] if url else list(self.extras)
+        requirement.extras = set() if url else set(self.extras)
```

Comment:
Non-blocking: according to L1191 `self.extras` is already referring to a set.

Reply:
Really, this should be `requirement.extras = set(self.extras) if not url`, because `requirement.extras` **is already** an empty set from `__init__()`, and it's not being assigned to anywhere else (that I can find).